templates-ci/php-component/github-actions/.github/workflows/deploy.yml

```diff
@@ -0,0 +1,45 @@
+name: Github actions
```

Comment:
Nepozdava sa mi to riesenie `deploy` a duplicitny kod.

Potrebujeme: spustit testy, ... a ak je tag `v.....`, tak nasledne aj `deploy`. Nevidim v tom 2 `workflows` ale iba jeden.

Okrem toho, mi tam chyba:
- Docker login, bud ho moze uzivatle doplnit dodatocne (vid podmienka `if` nizsie) alebo sa ho opytame v CLI.
  - Bez toho to moze padat.
- Chyba mi tam priprava na KBC testy, ... aby user uz iba vlozil token a ID konfiguracii ak chce.
- Neda sa to jednoducho paralelizovat, kedze `matrix` sa urcuje na urovni jobu a nie step. A nechces pre kazdu verziu databazy robit znova build.

**Navrhujem nieco taketo, mat to v jednom subore.**
Pre verziu bez PHP to iste, len vyhodit `composer ci` testy, a dat tam `echo "Here run your test tools."`

Mohli by sme zavolat a prebrat, ktore riesenie ma ake vyhody.

```yml
name: GitHub Actions
on: [ push ]
concurrency: ci-${{ github.ref }} # to avoid tag collisions in the ECR
env:
  # Name of the image in the ECR
  APP_IMAGE: keboola/db-extractor-pgsql
  # Developer portal login
  KBC_DEVELOPERPORTAL_VENDOR: "..."
  KBC_DEVELOPERPORTAL_APP: "..."
  KBC_DEVELOPERPORTAL_USERNAME: "..."
  KBC_DEVELOPERPORTAL_PASSWORD: ${{ secrets.KBC_DEVELOPERPORTAL_PASSWORD }}
  # DockerHub login
  DOCKERHUB_USER: "keboolabot"
  DOCKERHUB_TOKEN: ${{ secrets.DOCKERHUB_TOKEN }}
  # Test KBC project
  KBC_STORAGE_TOKEN: ${{ secrets.KBC_STORAGE_TOKEN }}
  KBC_TEST_PROJECT_URL: ""
  KBC_TEST_PROJECT_CONFIGS: "" # space separated list
jobs:
  build:
    runs-on: ubuntu-latest
    outputs:
      app_image_tag: ${{ steps.tag.outputs.app_image_tag }}
      is_semantic_tag: ${{ steps.tag.outputs.is_semantic_tag }}
    steps:
      - name: Check out the repo
        uses: actions/checkout@v2
      - name: Print Docker version
        run: |
          docker -v
      - name: Docker login
        if: env.DOCKERHUB_TOKEN
        run: docker login --username "$DOCKERHUB_USER" --password "$DOCKERHUB_TOKEN"
      - name: Build image
        run: docker build -t $APP_IMAGE .
      - name: Set image tag
        id: tag
        run: |
          TAG="${GITHUB_REF##*/}"
          IS_SEMANTIC_TAG=$(echo "$TAG" | grep -q '^v\?[0-9]\+\.[0-9]\+\.[0-9]\+$' && echo true || echo false)
          echo "Tag = '$TAG', is semantic tag = '$IS_SEMANTIC_TAG'"
          echo "::set-output name=app_image_tag::$TAG"
          echo "::set-output name=is_semantic_tag::$IS_SEMANTIC_TAG"
      - name: Push image to ECR
        uses: keboola/action-push-to-ecr@master
        with:
          vendor: ${{ env.KBC_DEVELOPERPORTAL_VENDOR }}
          app_id: ${{ env.KBC_DEVELOPERPORTAL_APP }}
          username: ${{ env.KBC_DEVELOPERPORTAL_USERNAME }}
          password: ${{ env.KBC_DEVELOPERPORTAL_PASSWORD }}
          tag: ${{ steps.tag.outputs.app_image_tag }}
          push_latest: ${{ steps.tag.outputs.is_semantic_tag }}
          source_image: ${{ env.APP_IMAGE}}

  tests:
    needs: build
    runs-on: ubuntu-latest
    # strategy:
    #   fail-fast: false
    #   matrix: ...
    steps:
      - name: Check out the repo
        uses: actions/checkout@v2
      - name: Pull image from ECR
        uses: keboola/action-pull-from-ecr@master
        with:
          vendor: ${{ env.KBC_DEVELOPERPORTAL_VENDOR }}
          app_id: ${{ env.KBC_DEVELOPERPORTAL_APP }}
          username: ${{ env.KBC_DEVELOPERPORTAL_USERNAME }}
          password: ${{ env.KBC_DEVELOPERPORTAL_PASSWORD }}
          tag: ${{ needs.build.outputs.app_image_tag }}
          target_image: ${{ env.APP_IMAGE}}
          tag_as_latest: true
      - name: Run tests
        run: docker run ${{env.APP_IMAGE}} composer ci

  tests-in-kbc:
    needs: build
    runs-on: ubuntu-latest
    if: env.KBC_STORAGE_TOKEN && env.KBC_TEST_PROJECT_CONFIGS
    steps:
      - name: Run KBC test jobs
        uses: keboola/action-run-configs-parallel@master
        with:
          token: ${{ env.KBC_STORAGE_TOKEN }}
          componentId: ${{ env.KBC_DEVELOPERPORTAL_APP }}
          tag: ${{ needs.build.outputs.app_image_tag }}
          configs: ${{ env.KBC_TEST_PROJECT_CONFIGS }}

  deploy:
    needs:
      - tests
      - tests-in-kbc
    runs-on: ubuntu-latest
    if: startsWith(github.ref, 'refs/tags/') && needs.build.outputs.is_semantic_tag == 'true'
    steps:
      - name: Set tag in the Deloper Portal
        uses: keboola/action-set-tag-developer-portal@master
        with:
          vendor: ${{ env.KBC_DEVELOPERPORTAL_VENDOR }}
          app_id: ${{ env.KBC_DEVELOPERPORTAL_APP }}
          username: ${{ env.KBC_DEVELOPERPORTAL_USERNAME }}
          password: ${{ env.KBC_DEVELOPERPORTAL_PASSWORD }}
          tag: ${{ needs.build.outputs.app_image_tag }}

```

Reply:
jj takhle už by to mohlo být fajn.... mě v tom řešení na mysql/pgsql vadila nepřehlednost, ale tohle mi příjde taky fajn 👍 

upravím to a vyzkouším

Reply:
upravil jsem to + jsem přidal možnost zadání credentials na Docker